### ISSUE TITLE
Support for RESTful URLs

### ISSUE THREAD

**Issue by darvid** [lynndylanhurley/ng-token-auth#122]
I know sessions and user authentication in general are inherently not RESTful, but it would be nice to have optional support for changing the method and allowing dynamic URLs with this module. For example:

```
Register: PUT /account/<username>
Login:    POST /session
Logout:   DELETE /session
Validate: GET /session
```

**Comment by booleanbetrayal** [comment]
@darvid - the logout path (`signOutUrl`) is called with `DELETE`, the sign-in path (`emailSignInPath`) is called with `POST`, the validate path (`tokenValidationPath`) is called with `GET` and the email registration (`emailRegistrationPath`) uses **`POST`**, since that is the expected REST paradigm for new record creation (whereas PUT generally implies knowledge of location / previous existence). 

In short, you should be able to configure this to be as REST-ful as you'd like. Hope this helps!
